### ISSUE TITLE
Unsuccessful update using MiKTeX console - seems to be a .dll or administrator privileges issue

### ISSUE THREAD

**Issue by rpinto73** [MiKTeX/miktex#256]
I am running MiKTeX Console 2.9.6966 with admin privileges on a computer with multiple Windows accounts.  I'm also using TeXworks v0.6.2 (MiKTeX 2,9,6960 64-bit). My computer uses Windows 7.

I just tried updating my packages using MiKTeX and it gave me an error while updating.  This is a copy of the error log it copied to the clipboard after giving me the update error message:

> MiKTeX: 2.9 OS: Windows 7 Professional, 64-bit, Service Pack 1, build
> 7601 SharedSetup: yes PathOkay: yes SystemAdmin: yes RootPrivileges:
> yes Root0: C:\Users\Rebecca\AppData\Roaming\MiKTeX\2.9 Root1:
> C:\Users\Rebecca\AppData\Local\MiKTeX\2.9 Root2:
> C:\ProgramData\MiKTeX\2.9 Root3: C:\Program Files\MiKTeX 2.9
> UserInstall: C:\Users\Rebecca\AppData\Roaming\MiKTeX\2.9 UserConfig:
> C:\Users\Rebecca\AppData\Roaming\MiKTeX\2.9 UserData:
> C:\Users\Rebecca\AppData\Local\MiKTeX\2.9 CommonInstall: C:\Program
> Files\MiKTeX 2.9 CommonConfig: C:\ProgramData\MiKTeX\2.9 CommonData:
> C:\ProgramData\MiKTeX\2.9 Invokers: explorer

Here is the report with info about my MiKTeX setup.
> Date: 2019-01-28 17:36:17 MiKTeX: 2.9.6960 OS: Windows 7 Professional,
> 64-bit, Service Pack 1, build 7601 SharedSetup: yes PathOkay: yes
> LastUpdateCheck: 2019-01-28 17:19:27 LastUpdate: not yet SystemAdmin:
> yes RootPrivileges: yes AdminMode: yes Root0:
> C:\ProgramData\MiKTeX\2.9 Root1: C:\Program Files\MiKTeX 2.9
> CommonInstall: C:\Program Files\MiKTeX 2.9 CommonConfig:
> C:\ProgramData\MiKTeX\2.9 CommonData: C:\ProgramData\MiKTeX\2.9
> 
> Warning: the following problems were detected:   1: installation is
> not up-to-date

I restarted MiKTeX and tried checking for updates but it said no updates were found.

Here are the two relevant log files:
[miktex-console_admin.log](https://github.com/MiKTeX/miktex/files/2805375/miktex-console_admin.log)
[miktex-report.txt](https://github.com/MiKTeX/miktex/files/2805376/miktex-report.txt)

Additionally, when I tried using TeXworks to compile my document, the console output shows:

> texify: warning: running with administrator privileges initexmf:
> warning: Option --admin should be specified when running this program
> with administrator privileges initexmf: warning: Option --admin should
> be specified when running this program with administrator privileges
> 
> Sorry, but "MiKTeX Compiler Driver" did not succeed.
> 
> The log file hopefully contains the information to get MiKTeX going
> again:
> 
>   C:\Users\Rebecca\AppData\Local\MiKTeX\2.9\miktex\log\texify.log

The texify.log file is attached:
> 2019-01-27 10:42:00,681-0500 WARN  texify - running with administrator privileges
[texify.log](https://github.com/MiKTeX/miktex/files/2805356/texify.log)

This is the error message when I try to run TeXworks:
![image](https://user-images.githubusercontent.com/15143365/51872778-aaa33f00-2328-11e9-9342-8989e71c1722.png)

And this is the console output from TeXworks:
![image](https://user-images.githubusercontent.com/15143365/51873106-d83cb800-2329-11e9-8788-777d3829e8e2.png)

This is all occuring on my school computer and it's a multiuser setup on Windows 7.  I only seem to have the option to run MiKTeX with administrator privileges.  At home, I also run MiKTeX 2.9.6996 (same version) on Windows 10 and I have the option to run in user-mode or administrator mode.  Not sure if this is causing issues.  Here is a screenshot of my Overview page.
![image](https://user-images.githubusercontent.com/15143365/51873378-e7703580-232a-11e9-9a00-e70549f830bd.png)

ps. This is my first post, if I'm missing information, please let me know.
I also tried to check for duplicate questions, but if I am duplicating a question, I would appreciate being pointed to the thread. Thanks.

**Comment by edocevoli** [comment]
Thank you for the report. Sorry, but it seems that it is not possible to run MiKTeX on your Windows system.

**Comment by edocevoli** [comment]
Looking at the log files, it seems that some application files are not up-to-date. Please open a command-prompt (as administrator) and run:

```
for /f "usebackq tokens=*" %f in (`where pdflatex.exe`) do dir "%f" >> output.txt
```

Please attach `output.txt`.

**Comment by rpinto73** [comment]
I tried running that from the Command Prompt (as an administrator) but it didn't seem to work.  When I opened TeXworks, it crashed and gave me the following error: 

![image](https://user-images.githubusercontent.com/15143365/51939174-cf58ee80-23dc-11e9-9e1b-087369a38dd6.png)

I tried going to MiKTeX again and updating it just now (it seems to have updated successfully), but TeXworks is still crashing and giving me the above error message.

Should I just give up and uninstall + reinstall MiKTeX and TeXworks?

**Comment by edocevoli** [comment]
Sorry, the command must include the `--force` option. So here is the correct command:

```
initexmf --admin --force --mklinks
```

**Comment by rpinto73** [comment]
That doesn't seem to work either. Here is the command prompt message
![image](https://user-images.githubusercontent.com/15143365/51942486-2f539300-23e5-11e9-8e7b-a2f31ce55462.png)

**Comment by rpinto73** [comment]
[initexmf_admin.log](https://github.com/MiKTeX/miktex/files/2809841/initexmf_admin.log)
Here is the initexmf_admin.log file

**Comment by rpinto73** [comment]
Update: I've just uninstalled and reinstalled MiKTeX on my system.  I uninstalled it using the MiKTeX console, and then I manually went into the following directories and deleted all the MiKTeX folders before reinstalling it.

> C:\ProgramData
> C:\Program Files
> C:\Users\Rebecca\AppData\Roaming
> C:\Users\Rebecca\AppData\Local

When I reinstalled it, I chose to install it only for me (private user, not system-wide).

This all seemed to install correctly.  When I tried opening TeXworks, it all was still fine.  However, as soon as I opened my main TeX file, I got the error I posted above regarding the graphical user interface to the typesetting system...has stoped working.

Is this now a TeXworks issue and no longer a MiKTeX issue?  I should specify that I have MiKTeX on my home computer and all my files (including that main TeX file) are compiling correctly.  So it's something on my school computer and I'm not sure if it's MiKTeX anymore or TeXworks.

Any advice?

**Comment by edocevoli** [comment]
Without seeing the log files it is difficult to identify the problem.

Have you tried to run MiKTeX/TeXworks with the virus scanner disabled (for testing purposes)?

**Comment by rpinto73** [comment]
My anti-virus is password protected by our department so I've sent a request for them to disable my antivirus.  In the mean time I have attached the log files here in case that gives you any clues. Thanks.

[miktex-console.log](https://github.com/MiKTeX/miktex/files/2812576/miktex-console.log)
[miktex-fc-cache.log](https://github.com/MiKTeX/miktex/files/2812577/miktex-fc-cache.log)
[mkfntmap.log](https://github.com/MiKTeX/miktex/files/2812578/mkfntmap.log)
[mpmcli.log](https://github.com/MiKTeX/miktex/files/2812579/mpmcli.log)
[texworks.log](https://github.com/MiKTeX/miktex/files/2812580/texworks.log)
[initexmf.log](https://github.com/MiKTeX/miktex/files/2812581/initexmf.log)

**Comment by edocevoli** [comment]
Thank you. The immediate workaround would be to remove links before running the update process:

```
initexmf --force --rmlinks
```

This will delete `pdflatex.exe` (and others), so that the updater will succeed (creating new executables).

**Comment by rpinto73** [comment]
Hi Christian,
I have disabled (uninstalled) the antivirus on my computer and I was still getting the error regarding my graphical user interface not working.

I tried your workaround and got the following output on Command Prompt, which was confusing, since it both asked me to write "--admin" in the command and also told me I shouldn't write it in the command.

![image](https://user-images.githubusercontent.com/15143365/52018278-d6a0fa80-24b7-11e9-8b6c-270507efc280.png)

I tried updating MiKTeX after running the command prompt and restarting TeXworks, and I'm still getting the same error regarding my graphical user interface not working.

I'm not sure if you have any more ideas?  I'm definitely frustrated because I'm not sure what the issue is.

Here are the log files after doing all of those operations by the way.
[miktex-console.log](https://github.com/MiKTeX/miktex/files/2814854/miktex-console.log)
[miktex-fc-cache.log](https://github.com/MiKTeX/miktex/files/2814855/miktex-fc-cache.log)
[mkfntmap.log](https://github.com/MiKTeX/miktex/files/2814856/mkfntmap.log)
[mpmcli.log](https://github.com/MiKTeX/miktex/files/2814857/mpmcli.log)
[texworks.log](https://github.com/MiKTeX/miktex/files/2814858/texworks.log)
[initexmf.log](https://github.com/MiKTeX/miktex/files/2814859/initexmf.log)

**Comment by edocevoli** [comment]
The `--admin` option must be omitted if this is a private installation. `texworks.exe` fails because this executable is also broken. On the other hand, `miktex-texworks.exe` should work. To repair your private MiKTeX installation, run:

```
initexmf --force --remove-links
initexmf --force --mklinks
```

You can verify your setup by comparing `texworks.exe` and `miktex-texworks.exe`. Both files must be identical.

**Comment by rpinto73** [comment]
When you say "You can verify your setup by comparing texworks.exe and miktex-texworks.exe. Both files must be identical."  How do you compare them?  I tried opening the files using Notepad++ but it's gibberish.

The size of the files look identical: 6 075 KB

Thanks in advance.

**Comment by edocevoli** [comment]
You can use `comp`:

```
cd "%LOCALAPPDATA%\Programs\MiKTeX 2.9\miktex\bin\x64"
comp /M pdflatex.exe miktex-pdflatex.exe
comp /M texworks.exe miktex-texworks.exe
```

**Comment by rpinto73** [comment]
Hi Christian,

I did that and the files compare ok.

I was still getting the graphical error from TeXworks everytime I opened my main .tex file, so I decided to try deleting the associated .pdf file and .synctex.gz file..... and lo and behold it compiled beautifully after that!  Problem solved!  

I'm not sure if I should Accept or Vote Up any of your replies on here?  Let me know what you think, since I think my technical issues changed over the course of this thread from problems with the .dll to just an issue with one of my local TeXworks associated files.  Since I also uninstalled and reinstalled MiKTeX twice, I'm not sure how that played into resolving any of this.

Thank you so much for your help and patience!  I really appreciate it and will gladly vote up or accept any of your answers if you think that's a good idea.
